File: app/src/main/java/com/owncloud/android/ui/trashbin/TrashbinRepository.java
/*
 * Nextcloud Android client application
 *
 * @author Tobias Kaminsky
 * Copyright (C) 2018 Tobias Kaminsky
 * Copyright (C) 2018 Nextcloud GmbH.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
package com.owncloud.android.ui.trashbin;

import com.owncloud.android.lib.resources.trashbin.model.TrashbinFile;

import java.util.List;

/**
 * Contract between presenter and model
 */
public interface TrashbinRepository {
    interface LoadFolderCallback {
        void onSuccess(List<Object> files);

        void onError(int error);
    }

    interface OperationCallback {
        void onResult(boolean success);
    }

    void getFolder(String remotePath, LoadFolderCallback callback);

    void restoreFile(TrashbinFile file, OperationCallback callback);

    void emptyTrashbin(OperationCallback callback);

    void removeTrashbinFile(TrashbinFile file, OperationCallback callback);
}

File: app/src/main/java/com/owncloud/android/ui/trashbin/TrashbinRepository.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2018 Tobias Kaminsky <tobias@kaminsky.me>
 * SPDX-FileCopyrightText: 2018 Nextcloud GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.ui.trashbin

import com.owncloud.android.lib.resources.trashbin.model.TrashbinFile

/**
 * Contract between presenter and model
 */
interface TrashbinRepository {
    interface LoadFolderCallback {
        fun onSuccess(files: List<TrashbinFile?>?)
        fun onError(error: Int)
    }

    interface OperationCallback {
        fun onResult(success: Boolean)
    }

    fun getFolder(remotePath: String?, callback: LoadFolderCallback?)
    fun restoreFile(file: TrashbinFile?, callback: OperationCallback?)
    fun emptyTrashbin(callback: OperationCallback?)
    fun removeTrashbinFile(file: TrashbinFile?, callback: OperationCallback?)
}
